os/pm/pm_procfs.c

```diff
@@ -481,7 +480,13 @@ static size_t power_lock_write(FAR struct file *filep, FAR const char *buffer, s
 	*/
 	pm_stay(PM_IDLE_DOMAIN, PM_NORMAL);
 	fvdbg("State locked!\n");
-
+	if (buflen > 0) {
+		g_pm_timer.use_timer = USE_PM_LOCK_TIMER;
```

Comment:
This method of assigning variable directly, can only support for current situation, if in the future, there is more request for timer usage, how do you plan to handle it? How about we use the bit masking method now?

Reply:
g_pm_timer.use_timer |= (1 << PM_LOCK_TIMER);

Reply:
even if we support multiple timers (more than 2), and we allow bit wise setting to indicate which timer to use,
The timer_interval member of the struct can still only accommodate only only value, for the latest timer setting that is called.
Hence, to avoid confusion with the bit wise masking, I am setting the timer type directly.

Reply:
Since we are setting the variable to a flat value, I believe we are going to have another callback handler for simultaneous timer operation, else we cannot identify to execute which operation when timer expires (in multiple timer scenario). Or do you have any other opinion to achieve this?
We should list out all possible ways to achieve the target, and choose the most efficient way with least overhead.

Reply:
> even if we support multiple timers (more than 2), and we allow bit wise setting to indicate which timer to use, The timer_interval member of the struct can still only accommodate only only value, for the latest timer setting that is called. Hence, to avoid confusion with the bit wise masking, I am setting the timer type directly.

We can add one more timer_interval to differentiate both one-shot and periodical, if both will run simultaneously.

Reply:
Working according to the current requirement of 2 types of timer callbacks (lock release and wakeup),
1. power_lock_write() can reset old PM_WAKEUP_TIMER. It will add bits to indicate that lock timer needs to be set, but it also needs to undo bits of wakeup timer or otherwise, in pg_int_handler callback, both timers will be found as set.
2. We have confirmation that both these timers currently cannot be invoked simultaneously because they are strictly associated with their own PM_STATEs which are mutually exclusive.
3. I am not considering about periodical timer as an additional functionality right now. From our previous discussion, we found that it can be very useful for keep alive requirements, but it comes with multiple considerations for multi-timer synchronization. My design ideas currently do not consider the scope of periodic timer in addition to one shout.
We can take it up if and when we really feel the need of an additional timer.